editor.htmlui/src/org/netbeans/modules/editor/htmlui/JSEmbeddingProvider.java

```diff
@@ -0,0 +1,191 @@
+import org.netbeans.api.java.source.JavaSource;
+import org.netbeans.api.lexer.Language;
+import org.netbeans.api.lexer.TokenHierarchy;
+import org.netbeans.api.lexer.TokenSequence;
+import org.netbeans.modules.parsing.api.Snapshot;
+import org.netbeans.modules.parsing.spi.Parser;
+import org.netbeans.modules.parsing.spi.Scheduler;
+import org.netbeans.modules.parsing.spi.SchedulerEvent;
+import org.netbeans.modules.parsing.spi.SchedulerTask;
+import org.netbeans.modules.parsing.spi.TaskFactory;
+
+/**
+ *
+ * @author Tomas Zezula
+ */
+public final class JSEmbeddingProvider extends JavaParserResultTask<Parser.Result> {
+    private static final Logger LOG = Logger.getLogger(JSEmbeddingProvider.class.getName());
+
+    private static final int PRIORITY = 1000;
+    private static final String JS_ANNOTATION = "net.java.html.js.JavaScriptBody";  //NOI18N
+    private static final String BODY = "body";                            //NOI18N
+    private static final String JAVA_MIME_TYPE = "text/x-java";           //NOI18N
+    private static final String JAVASCRIPT_MIME_TYPE = "text/javascript"; //NOI18N
+    private final AtomicBoolean canceled = new AtomicBoolean();
+
+    private JSEmbeddingProvider() {
+        super(JavaSource.Phase.ELEMENTS_RESOLVED);
+    }
+
+    @Override
+    public int getPriority() {
+        return PRIORITY;
+    }
+
+    @Override
+    public void cancel() {
+        canceled.set(true);
+    }
+
+    @Override
+    public Class<? extends Scheduler> getSchedulerClass() {
+        return Scheduler.EDITOR_SENSITIVE_TASK_SCHEDULER;
+    }
+
+    @Override
+    public void run(Parser.Result t, SchedulerEvent se) {
+        canceled.set(false);
+        final CompilationInfo ci = CompilationInfo.get(t);
+        final CompilationUnitTree cu = ci.getCompilationUnit();
+        final Trees trees = ci.getTrees();
+        final SourcePositions sp = trees.getSourcePositions();
+        final Finder f = new Finder(trees);
+        final List<LiteralTree> result = new ArrayList<LiteralTree>();
+        f.scan(cu, result);
+        if (!result.isEmpty()) {
+            try {
+                final TokenHierarchy<Document> tk = TokenHierarchy.get(ci.getDocument());
+                final Language<?> java = Language.find(JAVA_MIME_TYPE);
+                final Language<?> javaScript = Language.find(JAVASCRIPT_MIME_TYPE);
+                if (java != null && javaScript != null) {
+                    final TokenSequence<?> seq = tk.tokenSequence(java);
+                    if (seq != null) {
+                        for (LiteralTree lt : result) {
+                            final int start = (int) sp.getStartPosition(cu, lt);
+                            final int end = (int) sp.getEndPosition(cu, lt);
+                            seq.move(start);
+                            while (seq.moveNext() && seq.offset() < end) {
+                                seq.createEmbedding(javaScript, 1, 1, true);
```

Comment:
This is called, but even if the JavaScript support is installed, I don't see any coloring in the `body="...."` section. Can you advice what is wrong, @tzezula, thanks.